api/src/test/java/org/cardanofoundation/rosetta/api/network/controller/NetworkApiDelegateImplTest.java

```diff
@@ -0,0 +1,89 @@
+
+    @MockBean
+    private NetworkService networkService;
+
+    @Test
+    void networkList_Test() throws Exception {
+        //given
+        MetadataRequest metadataRequest = givenMetadataRequest();
+        NetworkListResponse networkListResponse = givenNetworkListResponse();
+        when(networkService.getNetworkList(any())).thenReturn(networkListResponse);
+        objectMapper.disable(SerializationFeature.FAIL_ON_EMPTY_BEANS);
+        //when
+        //then
+        mockMvc.perform(post("/network/list")
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(objectMapper.writeValueAsString(metadataRequest)))
+                .andDo(print())
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.network_identifiers[0].blockchain").value("cardano"))
+                .andExpect(jsonPath("$.network_identifiers[0].network").value("devkit"));
+    }
+
+    @Test
+    void networkStatus_Test() throws Exception {
+        //given
+        NetworkRequest networkRequest = givenNetworkRequest();
+        NetworkStatusResponse networkStatusResponse = givenNetworkStatusResponse();
+        when(networkService.getNetworkStatus(any())).thenReturn(networkStatusResponse);
+        objectMapper.disable(SerializationFeature.FAIL_ON_EMPTY_BEANS);
+        //when
+        //then
+        mockMvc.perform(post("/network/status")
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(objectMapper.writeValueAsString(networkRequest)))
+                .andDo(print())
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.current_block_identifier.index").value(123L))
+                .andExpect(jsonPath("$.current_block_identifier.hash").value("123"));
+    }
+
+    @Test
+    void networkOptions_Test() throws Exception {
+        //given
+        NetworkRequest networkRequest = givenNetworkRequest();
+        NetworkOptionsResponse networkOptionsResponse = givenNetworkOptionsResponse();
+        when(networkService.getNetworkOptions(any())).thenReturn(networkOptionsResponse);
+        objectMapper.disable(SerializationFeature.FAIL_ON_EMPTY_BEANS);
+        //when
+        //then
+        mockMvc.perform(post("/network/options")
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(objectMapper.writeValueAsString(networkRequest)))
+                .andDo(print())
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.version.rosetta_version").value(networkOptionsResponse.getVersion().getRosettaVersion()))
+                .andExpect(jsonPath("$.version.node_version").value(networkOptionsResponse.getVersion().getNodeVersion()))
+                .andExpect(jsonPath("$.allow.operation_statuses[0].status").value(networkOptionsResponse.getAllow().getOperationStatuses().getFirst().getStatus()));
+    }
+
```

Comment:
Negative tests to add:
https://github.com/cardano-foundation/cardano-rosetta-java/pull/162/files#diff-2a3ca67d67d5afa48e6c9518787c10fb7dc02b01514ed6ae87faa72db8d8af8eR104-R162

Reply:
I'll do it in the next pull request.